File: Java/Motherboard.java
public class Motherboard extends Komponen {
    private String chipset;
    private String formFactor;
    private String socketType;
    
    public Motherboard() {
        super();
        this.chipset = "";
        this.formFactor = "";
        this.socketType = "";
    }
    
    public Motherboard(String chipset, String formFactor, String socketType, 
                      String merk, String nama, double harga) {
        super(merk, nama, harga, "Motherboard");
        this.chipset = chipset;
        this.formFactor = formFactor;
        this.socketType = socketType;
    }
    
    public String getChipset() {
        return chipset;
    }
    
    public void setChipset(String chipset) {
        this.chipset = chipset;
    }
    
    public String getFormFactor() {
        return formFactor;
    }
    
    public void setFormFactor(String formFactor) {
        this.formFactor = formFactor;
    }
    
    public String getSocketType() {
        return socketType;
    }
    
    public void setSocketType(String socketType) {
        this.socketType = socketType;
    }
    
    @Override
    public void tampilkanInfo() {
        System.out.println("Motherboard: " + merk + " " + nama + 
                           " (" + chipset + ", " + formFactor + ")");
        System.out.println("    Socket: " + socketType + ", Harga: Rp" + harga);
    }
}

File: Python/Motherboard.py
# motherboard.py
from Komponen import Komponen

class Motherboard(Komponen):
    """Motherboard class"""
    
    def __init__(self, chipset="", form_factor="", socket_type="", merk="", nama="", harga=0, tipe_komponen="Motherboard"):
        super().__init__(merk, nama, harga, tipe_komponen)
        self.chipset = chipset
        self.form_factor = form_factor
        self.socket_type = socket_type
    
    def get_chipset(self):
        return self.chipset
    
    def set_chipset(self, chipset):
        self.chipset = chipset
    
    def get_form_factor(self):
        return self.form_factor
    
    def set_form_factor(self, form_factor):
        self.form_factor = form_factor
    
    def get_socket_type(self):
        return self.socket_type
    
    def set_socket_type(self, socket_type):
        self.socket_type = socket_type
    
    def tampilkan_info(self):
        print(f"Motherboard: {self.merk} {self.nama} ({self.chipset}, {self.form_factor})")
        print(f"    Socket: {self.socket_type}, Harga: Rp{self.harga}")